pkg/cache/cluster.go

```diff
@@ -62,10 +61,24 @@ func list(resClient dynamic.ResourceInterface, opts metav1.ListOptions) (*unstru
 
 type apiMeta struct {
 	namespaced      bool
-	resourceVersion string
+	resourceVersion atomic.Value
 	watchCancel     context.CancelFunc
 }
 
+func (a *apiMeta) LoadResourceVersion() (string, error) {
+	v := a.resourceVersion.Load()
+	version, ok := v.(string)
+	if ok {
+		return version, nil
+	}
+
+	return "", fmt.Errorf("stored type is invalid in resourceVersion")
```

Comment:
This cannot happen and if it does it should be ok to panic because... it should never happen?
I'd do this instead

```go
str, _ := a.resourceVersion.Load().(string)
return str, nil
```
This will not error out and it also preserves current semantics where the field is an empty string if it was never set. It's also nicer to not need to check the error in every place where this method is called.